concrete-core/src/backends/core/implementation/engines/ggsw_ciphertext_discarding_encryption.rs

```diff
@@ -0,0 +1,161 @@
+use concrete_commons::dispersion::Variance;
+
+use crate::backends::core::implementation::engines::CoreEngine;
+use crate::backends::core::implementation::entities::{
+    GgswCiphertext32, GgswCiphertext64, GlweSecretKey32, GlweSecretKey64, Plaintext32, Plaintext64,
+};
+use crate::specification::engines::{
+    GgswCiphertextDiscardingEncryptionEngine, GgswCiphertextDiscardingEncryptionError,
+};
+
+/// # Description:
+/// Implementation of [`GgswCiphertextDiscardingEncryptionEngine`] for [`CoreEngine`] that operates
+/// on 32 bits integers.
+impl GgswCiphertextDiscardingEncryptionEngine<GlweSecretKey32, Plaintext32, GgswCiphertext32>
+    for CoreEngine
+{
+    /// # Example:
+    /// ```
+    /// use concrete_commons::dispersion::Variance;
+    /// use concrete_commons::parameters::{
+    ///     DecompositionBaseLog, DecompositionLevelCount, GlweDimension, PolynomialSize,
+    /// };
+    /// use concrete_core::prelude::*;
+    /// # use std::error::Error;
+    ///
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and are not secure.
+    /// let glwe_dimension = GlweDimension(2);
+    /// let polynomial_size = PolynomialSize(4);
+    /// let level = DecompositionLevelCount(1);
+    /// let base_log = DecompositionBaseLog(4);
+    /// // Here a hard-set encoding is applied (shift by 20 bits)
+    /// let input = 3_u32 << 20;
+    /// let noise = Variance(2_f64.powf(-25.));
+    ///
+    /// let mut engine = CoreEngine::new()?;
+    /// let key_1: GlweSecretKey32 = engine.create_glwe_secret_key(glwe_dimension, polynomial_size)?;
+    /// let plaintext = engine.create_plaintext(&input)?;
+    /// let mut ciphertext =
+    ///     engine.encrypt_ggsw_ciphertext(&key_1, &plaintext, noise, level, base_log)?;
+    /// // We're going to re-encrypt the input with another secret key
+    /// // For this, it is required that the second secret key uses the same GLWE dimension
+    /// // and polynomial size as the first one.
+    /// let key_2: GlweSecretKey32 = engine.create_glwe_secret_key(glwe_dimension, polynomial_size)?;
+    ///
+    /// engine.discard_encrypt_ggsw_ciphertext(&key_2, &mut ciphertext, &plaintext, noise)?;
+    /// #
+    /// assert_eq!(ciphertext.glwe_dimension(), glwe_dimension);
+    /// assert_eq!(ciphertext.polynomial_size(), polynomial_size);
+    ///
```

Comment:
There is an issue open regarding GGSW decryption, it will not be adressed in this PR. Also, I don't think we want to test that in the doctest, as you said.